src/connectors/menu/__tests__/connectMenu-test.js

```diff
@@ -617,144 +643,311 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/menu/js/#co
     });
   });
 
-  describe('routing', () => {
-    const getInitializedWidget = () => {
-      const rendering2 = jest.fn();
-      const makeWidget2 = connectMenu(rendering2);
-      const widget = makeWidget2({
-        attribute: 'category',
+  describe('getWidgetState', () => {
+    test('returns the `uiState` empty', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const helper = jsHelper({}, '');
+      const widget = makeWidget({
+        attribute: 'brand',
       });
 
-      const helper = jsHelper(
+      const actual = widget.getWidgetState(
         {},
-        '',
-        widget.getConfiguration(new SearchParameters())
+        {
+          searchParameters: helper.state,
+        }
       );
-      helper.search = jest.fn();
 
-      widget.init({
-        helper,
-        state: helper.state,
-        createURL: () => '#',
-      });
+      expect(actual).toEqual({});
+    });
 
-      const { refine } = rendering2.mock.calls[0][0];
+    test('returns the `uiState` with a refinement', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const helper = jsHelper({}, '', {
+        hierarchicalFacets: [
+          {
+            name: 'brand',
+            attributes: ['brand'],
+          },
+        ],
+        hierarchicalFacetsRefinements: {
+          brand: ['Apple'],
+        },
+      });
 
-      return [widget, helper, refine];
-    };
+      const widget = makeWidget({
+        attribute: 'brand',
+      });
 
-    describe('getWidgetState', () => {
-      test('should give back the object unmodified if there are no refinements', () => {
-        const [widget, helper] = getInitializedWidget();
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+      const actual = widget.getWidgetState(
+        {},
+        {
           searchParameters: helper.state,
-          helper,
-        });
+        }
+      );
 
-        expect(uiStateAfter).toBe(uiStateBefore);
+      expect(actual).toEqual({
+        menu: {
+          brand: 'Apple',
+        },
       });
+    });
 
-      test('should add an entry equal to the refinement', () => {
-        const [widget, helper] = getInitializedWidget();
-        helper.toggleRefinement('category', 'pants');
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
-          searchParameters: helper.state,
-          helper,
-        });
-
-        expect(uiStateAfter).toEqual({
-          menu: {
-            category: 'pants',
+    test('returns the `uiState` without namespace overridden', () => {
```

Comment:
not sure what this test name means

Reply:
The sentence is now used in a couple of places. We can update them all at once into a dedicated PR.